src/app/pages/products-list/card/card.component.ts

```diff
@@ -0,0 +1,18 @@
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {Product} from 'src/app/shared/products/product.interface';
+
+@Component({
+    selector: 'app-card',
+    templateUrl: './card.component.html',
+    styleUrls: ['./card.component.css'],
+})
+export class CardComponent {
+    @Input() product: Product | undefined;
+
+    @Output() buyClick = new EventEmitter<Product['_id']>();
```

Comment:
Рекомендую использовать отличную практику улучшающую качество кода и упращающую читаемость: помечать не изменяющиеся свойства как `readonly`
```suggestion
    @Output() readonly buyClick = new EventEmitter<Product['_id']>();
```